### PR TITLE
refactor: relocate context providers into their domain packages + consolidate prompt-format helpers

### LINKED ISSUE #625: refactor: relocate context providers into their domain packages
## Context

PR #624 placed the notification history context provider in `internal/notifications/` rather than `internal/awareness/`. This was a deliberate experiment: domain-local providers keep related code together and reduce the surface area of the centralized `awareness/` package.

The pattern worked well — `notifications.HistoryProvider` implements `agent.ContextProvider` and lives alongside the `RecordStore` it queries, with no cross-package plumbing beyond the interface.

However, PR #624 was a good pattern, not a complete separation. `history_provider.go` still reaches back into `awareness.FormatDeltaOnly` (`timefmt.go`), so "move provider into its domain package" must mean **provider plus its domain-specific formatting and lifecycle live there, with no back-edge into `awareness/`** except very small shared primitives.

## Definition of success

**No moved provider should import `internal/awareness` afterward.** This is real decomposition, not file relocation.

### Extract shared formatting helpers first

`FormatDelta` / `FormatDeltaOnly` in `awareness/timefmt.go` are already used outside `awareness/` (e.g., `notifications.HistoryProvider`). They should move to a tiny neutral package (e.g., `internal/timefmt` or `internal/prompt`) so `awareness/` doesn't become a utility drawer that every domain package imports.

This is the enabling step — without it, every relocated provider would retain a back-edge into `awareness/`.

## Candidates

| Current location | Proposed location | Notes |
|-----------------|-------------------|-------|
| `awareness.StateWindowProvider` | `homeassistant.StateWindowProvider` | **Best next candidate.** Already HA-owned: consumes HA state-change events, exposes `HandleStateChange` plus prompt rendering. Moving to `internal/homeassistant/` is a real simplification. |
| `awareness.ChannelOverviewProvider` | Stay or move to `agent/` | **Should not move just for symmetry.** Depends on loop registry, contact resolution, and request hints — it's a composite/assembly concern, not a single domain. May belong in `agent/` or remain as-is. |
| `awareness.WatchlistProvider` | Stay | Watchlist store is native to `awareness/` |
| `awareness.WatchlistTagProvider` | Stay | Same — watchlist is the domain |
| `contacts.ContextProvider` | Already domain-local ✓ | |
| `memory.EpisodicProvider` | Already domain-local ✓ | |
| `memory.WorkingMemoryProvider` | Already domain-local ✓ | |
| `memory.ArchiveContextProvider` | Already domain-local ✓ | |
| `scheduler.AnticipationProvider` | Already domain-local ✓ | |
| `knowledge.SubjectContextProvider` | Already domain-local ✓ | |
| `notifications.HistoryProvider` | Already domain-local ✓ | Established the pattern in #624 |

## What `awareness/` becomes

After this work, `awareness/` would contain the watchlist (store, provider, tag provider) and any truly shared primitives that didn't get extracted. If watchlist is the only true domain left, either let `awareness/` intentionally shrink to that single responsibility, or plan a rename later.

## Approach

1. Extract `FormatDelta` / `FormatDeltaOnly` to a neutral package (prerequisite)
2. Move `StateWindowProvider` to `internal/homeassistant/`
3. Evaluate `ChannelOverviewProvider` — decide whether it belongs in `agent/`, stays, or gets restructured
4. One provider per PR, each verified by `just ci`
5. Registration stays in `app/new_awareness.go`

## Prior art

- #624 — notification history provider (established the pattern)

### PR DESCRIPTION
**Pull request by thane-developer**
## Summary

Closes #625 and broadens the extracted time-formatting package into a
proper prompt-formatting primitives package.

### #625 — provider relocations

- `FormatDelta`, `FormatDeltaOnly`, `ParseTimeOrDelta` → new neutral
  package (now `internal/promptfmt`, see below). Eleven domain packages
  previously imported `awareness` only for these helpers; they now drop
  the awareness import outright.
- `StateWindowProvider` → [`internal/homeassistant`](internal/homeassistant/statewindow.go).
  HA owns its data path (consumes state-change events).
- `ChannelOverviewProvider` (+ `ChannelLoopSource`, `PhoneResolver`,
  `HintsFunc`, `LoopSnapshot`) → [`internal/agent`](internal/agent/channel_overview.go).
  It assembles from loop registry + contacts + request hints, so it's
  agent-level composition, not a domain primitive. Co-locates with the
  existing `agent.ChannelProvider`.

### Scope broadening — prompt-formatting primitives

On review, `internal/timefmt` was too narrow a name for the package it
was becoming. Surveyed the codebase and found four more
domain-agnostic helpers scattered as unexported copy-paste candidates:

| Was | Now | Role |
|---|---|---|
| `shortID` (agent) | `promptfmt.ShortIDPrefix` | rune-aware first-8 truncation |
| `truncateID` (awareness) | `promptfmt.ShortIDSuffix` | byte-based last-8 truncation |
| `formatNumber` (awareness) | `promptfmt.FormatNumber` | thousands-grouped integer |
| `marshalCompact` (awareness) | `promptfmt.MarshalCompact` | one-line JSON with fallback |

Renamed the package `internal/timefmt` → `internal/promptfmt` and
folded the four helpers in. All were unexported, so consolidation
doesn't change any previously-reachable API; new test coverage is
added for `ShortIDPrefix` (including a rune-boundary case) and
`MarshalCompact` (including the unmarshalable-input fallback). Package
doc comment explicitly scopes it to "presentation-layer helpers for
prompt and context strings" so it doesn't become a utility drawer.

## Definition-of-success check (from #625 body)

> No moved provider should import `internal/awareness` afterward.

Verified:

- `notifications.HistoryProvider` — ✓ (closes the back-edge #625 called out)
- `homeassistant.StateWindowProvider` — ✓
- `agent.ChannelOverviewProvider` — ✓

Surviving cross-package `awareness` imports are all legitimate
consumers of watchlist, entity formatting, `CurrentConditions`, or
`ContextUsageInfo` — not relocated-provider back-edges.

## What `awareness/` looks like now

Shrunk to its actual responsibilities:
- `conditions.go` — `CurrentConditions` prompt primitive
- `context_usage.go` — token-usage formatting (no longer owns the int/ID helpers)
- `entity_format.go` — HA entity JSON formatters (no longer owns marshalCompact)
- `watchlist_*.go` — watchlist store + providers

A future pass could split entity formatting into `homeassistant/`, but
that was not required by this issue.

## Commits

1. `refactor: extract relative-time helpers to internal/timefmt`
2. `refactor: relocate StateWindowProvider to homeassistant package`
3. `refactor: relocate ChannelOverviewProvider to agent package`
4. `chore: reorder timefmt import to satisfy gofmt`
5. `chore: bump architecture baseline for internal/timefmt package`
6. `refactor: rename internal/timefmt to internal/promptfmt`
7. `refactor: consolidate prompt-formatting helpers into promptfmt`

## Test plan

- [x] \`just ci\` passes locally on the final commit
- [ ] CI green on PR
- [x] No relocated provider imports \`internal/awareness\`
- [x] Surviving \`awareness\` imports are consumers of non-relocated
  primitives (watchlist, entity formatting, conditions, context usage)
- [x] New exported helpers have test coverage (\`ShortIDPrefix\`,
  \`ShortIDSuffix\`, \`FormatNumber\`, \`MarshalCompact\`)

### CODE REVIEW

File: internal/promptfmt/num.go [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+func FormatNumber(n int) string {
+	s := fmt.Sprintf("%d", n)
+	if len(s) <= 3 {
+		return s
+	}
+	var sb strings.Builder
+	remainder := len(s) % 3
+	if remainder > 0 {
+		sb.WriteString(s[:remainder])
+	}
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/promptfmt/num.go]: Fixed in f927a4e. Sign is stripped at the string level (via strconv.Itoa, not arithmetic negation, so math.MinInt is safe) and re-prepended after grouping. Added -1/-100/-999/-1000/-31200/-1234567 test cases; -100 specifically would have reproduced the reported "-,100" output.

File: internal/promptfmt/json.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+// MarshalCompact returns v as a single-line JSON string with no extra
+// whitespace. On marshal failure it falls back to a fmt.Sprintf %v
+// rendering so the prompt path always produces *some* text; forensic
+// records and tests can still detect the failure mode by matching on
+// the absence of JSON delimiters.
+func MarshalCompact(v any) string {
+	data, err := json.Marshal(v)
+	if err != nil {
+		return fmt.Sprintf("%v", v)
```
[the remark this answers is not in the document]
Comment (thane-developer) [reply internal/promptfmt/json.go]: Fixed in 26276f4. Fallback now returns `[json-marshal-error: <err>] %v` and the package exposes `HasMarshalError(s string) bool` so detectors check the sentinel rather than output shape. Added a struct-with-Inf regression test that would have failed under the old heuristic.
